### ISSUE TITLE
unfair setting

### ISSUE THREAD

**Issue by Fu-Zhou-2019** [YingYuChen97/CSCA#2]
If I understand correctly, in main.py, the steps of PT-MAP are used to extract new class features and apply them directly. Isn't this training on the test set, which I feel is an unfair comparison and should not involve any gradient update steps when testing new classes?

**Comment by YingYuChen97** [comment]
Only support sets are used for training in main.py.
Please reopen this issue if you got any problems.

**Comment by Fu-Zhou-2019** [comment]
`   with torch.no_grad():
            for i in range(opt.way):
                sim_matrix = [[] for _ in range(opt.way)]
                
                for j in range(opt.way):
                    output = model[i](mtest_data).squeeze(-1).squeeze(-1)
                    sim_matrix[j] = cos(output[i * opt.query:(i+1) * opt.query],prototypes[j]).cpu().data.numpy()
                
                pred = np.argmax(sim_matrix,axis=0)
                correct += np.count_nonzero(pred==i)`

In this code, you seem to have implied the query's  label. When the outer loop i is fixed, the inner loop send the query samples belonging to class i to model[i]. 
What about when a query sample that does not belong to  class i is sensed in the model[i]?
